doc/user-guide.md

```diff
@@ -11,189 +59,301 @@ funcool/rumext
+
+An example of how it can be used and combined with `mf/spread`:
 
 ```clojure
-(mf/defc title*
-  [{:keys [name] :rest props}]
-  (assert (object? props) "expected object")
-  (assert (nil? (unchecked-get props "name")) "no name in props")
+(mf/defc my-label*
+  [{:keys [name class on-click] :rest props}]
+  (let [class (or class "my-label")
+        new-props (mf/props {:class class})
+        all-props (mf/spread props new-props)]
+    [:span {:on-click on-click}
+      [:> :label props name]]))
+```
+
+The map, being a Clojure object, may also be created elsewhere and passed
+to `mf/props` as a variable:
 
-  ;; The `:>` will be explained later
+```clojure
+(let [clj-props {:class "my-label"}
+      props (mf/props clj-props)]
   [:> :label props name])
 ```
 
-This allows you to extract the props that the component has control of
-and leave the rest in an object that can be passed as-is to the next
-element.
+But in this case, `props` binding will contain a plain js object converted
+dinamically from clojure map at runtime, by using the `clj->js` function. This
+should be avoided if performance is important because it adds the overhead of
+dynamic conversion on each render.
 
-## JSX / Hiccup
+### Instantiating a custom component
 
-You may already be familiar with Hiccup syntax (which is equivalent to
-the React JSX) for defining the React DOM. The intention of this
-section is to explain only the essential part of it and the
-peculiarities of Rumext.
+You can pass to `:>` macro the name of a custom component (see [below](#creating-a-react-custom-component))
+to create an instance of it:
 
-Let's start with simple generic elements like `div`:
+```clojure
+(mf/defc my-label*
+  [{:keys [name class on-click] :rest props}]
+    [:span {:on-click on-click}
+      [:> :label props name]])
+
+(mf/defc other-component
+  []
+  [:> my-label* {:name "foobar" :on-click some-fn}])
+```
+
+## Creating a React custom component
+
+The `defc` macro is the basic block of a Rumext UI. It's a lightweight utility
+that generates a React **function component** and adds some adaptations for it
+to be more convenient to ClojureScript code, like `camelCase` conversions and
+reserved name changes as explained [above](#passing-props).
+
+For example, this defines a React component:
 
 ```clojure
-[:div {:class "foobar"
-       :style {:background-color "red"}
-       :on-click some-on-click-fn}
-  "Hello World"]
+(require '[rumext.v2 :as mf])
+
+(mf/defc title*
+  [{:keys [label-text] :as props}]
+  [:div {:class "title"} label-text])
 ```
 
-The props and the style are transformed at compile time into a JS
-object, transforming all keys from lisp-case to camelCase (and
-renaming `:class` to `className`); so the compilation results in
-something like this:
+The compiled javascript for this block will be similar to what would be
+obtained for this JSX block:
 
 ```js
-const h = React.createElement;
+export default function title({labelText}) {
```

Comment:
I suggest remove the `export default` from here. Is not relevant to the example and the const h = React...  because  it is unused on the example